app/src/models/multi-commit-operation.ts

```diff
@@ -216,3 +224,14 @@ export type MultiCommitOperationDetail =
   | ICherryPickDetails
   | IRebaseDetails
   | IMergeDetails
+
+export function instanceOfIBaseRebaseDetails(
+  object: any
+): object is IBaseInteractiveRebaseDetails {
+  return 'commits' in object && 'currentTip' in object
```

Comment:
Not a fan of this check 😕 

This is not much better, but what if we declare a dummy object of type `IBaseInteractiveRebaseDetails` and check if `object` contains at least all the properties of that dummy object?

That way, the compiler will force us to include all the properties from `IBaseInteractiveRebaseDetails` in the dummy object in case we add/remove any:

```suggestion
  const dummy: IBaseInteractiveRebaseDetails = {
    commits: [],
    currentTip: 'main',
  }

  return Object.keys(dummy).every(key => key in object)
```

Reply:
Yeah. This is kind of `meh` but I like your much better. But to note, it is the `): object is IBaseInteractiveRebaseDetails {` part that makes typescript happy.  I could put `return true` in there for all typescript cares.

Reply:
That's good to know, but it's better if the implementation matches the signature as closely as possible 😂

Reply:
Absolutely!